src/main/java/reposense/parser/GroupConfigCsvParser.java

```diff
@@ -0,0 +1,86 @@
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+
+import org.apache.commons.csv.CSVRecord;
+
+import reposense.model.FileType;
+import reposense.model.GroupConfiguration;
+import reposense.model.RepoLocation;
+
+/**
+ * Container for the values parsed from {@code groups-config.csv} file.
+ */
+public class GroupConfigCsvParser extends CsvParser<GroupConfiguration> {
+    public static final String GROUP_CONFIG_FILENAME = "groups-config.csv";
+
+    /**
+     * Positions of the elements of a line in groups-config.csv config file
+     */
+    private static final int LOCATION_POSITION = 0;
+    private static final int GROUP_NAME_POSITION = 1;
+    private static final int FILES_GLOB_POSITION = 2;
+
+    public GroupConfigCsvParser(Path csvFilePath) throws IOException {
+        super(csvFilePath);
+    }
+
+    /**
+     * Gets the list of positions that are mandatory for verification.
+     */
+    @Override
+    protected int[] mandatoryPositions() {
+        return new int[] {
+            LOCATION_POSITION, GROUP_NAME_POSITION, FILES_GLOB_POSITION,
+        };
+    }
+
+    /**
+     * Processes the csv file line by line and adds created {@code Group} into {@code results}.
+     */
+    @Override
+    protected void processLine(List<GroupConfiguration> results, CSVRecord record) {
+        String location = get(record, LOCATION_POSITION);
+        String groupName = get(record, GROUP_NAME_POSITION);
+        List<String> globList = getAsList(record, FILES_GLOB_POSITION);
+
+        GroupConfiguration groupConfig = null;
+        try {
+            groupConfig = findMatchingGroupConfiguration(results, location);
+        } catch (InvalidLocationException e) {
+            e.printStackTrace();
+        }
+
+        FileType group = new FileType(groupName, globList);
+        if (groupConfig.containsGroup(group)) {
+            logger.warning(String.format(
+                    "Skipping group as %s has already been specified for the repository %s",
+                    group.toString(), groupConfig.getLocation()));
+            return;
+        }
+
+        groupConfig.addGroup(group);
+    }
+
+    /**
+     * Gets an existing {@code GroupConfiguration} from {@code results} if {@code location} matches.
+     * Otherwise adds a newly created {@code GroupConfiguration} into {@code results} and returns it.
+     *
+     * @throws InvalidLocationException if {@code location} is invalid.
+     */
+    private static GroupConfiguration findMatchingGroupConfiguration(
+            List<GroupConfiguration> results, String location) throws InvalidLocationException {
+        GroupConfiguration config = new GroupConfiguration(new RepoLocation(location));
+
+        for (GroupConfiguration groupConfig : results) {
+            if (groupConfig.getLocation().equals(config.getLocation())) {
+                return groupConfig;
+            }
+        }
+
+        results.add(config);
+        return config;
+    }
```

Comment:
Not necessary in this PR.

Can make this a generic method but require creation of a new parent class.

Reply:
I don't quite get it why is it not necessary..

We need to find out if an existing `GroupConfiguration` (tagged to `RepoLocation`) exists in order to append the new groups in. If no existing `GroupConfiguration` exists we need to add a new `GroupConfiguration` into the existing list of `results` and return it.

And also a generic method to replace `findMatchingGroupConfiguration` may not be a good idea as the one in `AuthorConfigCsvParser#findMatchingAuthorConfiguration` has additional checks if the branch is equal (matching group doesn't need to check if the branch is equal or not).

Reply:
> And also a generic method to replace findMatchingGroupConfiguration may not be a good idea as the one in AuthorConfigCsvParser#findMatchingAuthorConfiguration has additional checks if the branch is equal (matching group doesn't need to check if the branch is equal or not).

Sorry I meant, not necessary to make a generic method now. Its doable, can always get your base classes to implement a boolean method for evaluation.

Reply:
Ohh I think I get what you mean now..

Maybe we can deal with this over another PR specifically to clean up codes?